src/or/dirauth/shared_random.h

```diff
@@ -101,11 +101,40 @@ typedef struct sr_commit_t {
 
 /* API */
 
-/* Public methods: */
+/* Public methods used _outside_ of the module.
+ *
+ * We need to nullify them if the module is disabled. */
+#ifdef HAVE_MODULE_DIRAUTH
```

Comment:
We should make sure we have "#ifdef HAVE_MODULE_DIRAUTH" surrounding all of the include files in dirauth.h, except for any parts that should be accessible with dirauths are disabled.

Reply:
err, whoops.  I meant "all the include files in dirauth/*.h"

Reply:
We can't ... header file in `dirauth/` have symbols used outside of the dirauth module which are NOP if no dirauth module.

If we ifdef the include statement, it will fail to compile by not finding the symbols.